lib/matplotlib/tests/test_figure.py

```diff
@@ -1770,6 +1770,44 @@ def test_warn_colorbar_mismatch():
         subfig3_1.colorbar(im4_1)
 
 
+def test_clf_subplotpars():
+    keys = ('left', 'right', 'bottom', 'top', 'wspace', 'hspace')
+    rc_params = {key: plt.rcParams['figure.subplot.'+key] for key in keys}
+
+    fig = plt.figure(1)
+    fig.subplots_adjust(left=0.1)
+    fig.clf()
+    assert fig.subplotpars.get_subplot_params() == rc_params
+
+
+def test_suplots_adjust_1():
+    fig = plt.figure(1)
+    wspace = 0
+    fig.subplots_adjust(wspace=wspace)
+    inDict = dict(left=0.1, right=0.7, bottom=0, top=0.9, hspace=0.05)
+    fig.subplots_adjust(**inDict)
+    inDict['wspace'] = wspace
+    assert fig.subplotpars.get_subplot_params() == inDict
```

Comment:
What are you testing here? Is this just that the not-given parameters are not modified? If so, I'd write this as

```suggestion
def test_suplots_adjust_incremental():
    fig = plt.figure()
    fig.subplots_adjust(left=0)
    fig.subplots_adjust(right=1)
    assert fig.subplotpars.left == 0
    assert fig.subplotpars.right == 1
```